File: server/models/BlogPost.js
// models/BlogPost.js
const mongoose = require('mongoose');

const blogPostSchema = new mongoose.Schema({
  title: { type: String, required: true },
  excerpt: String,
  coverImage: String,
  category: String,
  author: {
    name: String,
    avatar: String,
    title: String, // optional, if you want to include a title for the author
  },
  date: { type: Date, default: Date.now },
  readTime: Number,
  tags: [String],
  externalLink: String,
}, { timestamps: true });

module.exports = mongoose.model('BlogPost', blogPostSchema);

File: server/models/blog_post.py
import datetime
from mongoengine import Document, StringField, DateTimeField, ListField, EmbeddedDocument, EmbeddedDocumentField, IntField

class Author(EmbeddedDocument):
    name = StringField(required=True)
    avatar = StringField()
    title = StringField()

class BlogPost(Document):
    title = StringField(required=True)
    excerpt = StringField()
    coverImage = StringField()
    category = StringField()
    author = EmbeddedDocumentField(Author)
    date = DateTimeField(default=datetime.datetime.utcnow)
    readTime = IntField()
    tags = ListField(StringField())
    externalLink = StringField()

    meta = {'collection': 'blog_posts'}
